File: 0671.cpp
#include <iostream>
#include <vector>
#include <cmath>

int z;					// длина строки/числа
std::string str;		// входящая строка числа
long long int s = 0;	// сумма чисел из 4 и 7
std::vector<int> arr;	// цифры числа

// рекурсия
long long int rec(int pos, bool same) {
	if (!same)
		return (pow(2, (z - pos)));
	if (z == pos)
		return 1;
	long long int t = 0;
	if (4 < arr[pos])
		t += rec(pos + 1, 0);
	if (7 < arr[pos])
		t += rec(pos + 1, 0);
	if (4 == arr[pos] || 7 == arr[pos])
		t += rec(pos + 1, 1);
	return t;
}

int main() {
	std::cin >> str;
	z = size(str);

	for (char const& c : str)
		arr.push_back(c-'0');

	for (int i = 1; i < z; i++)
		s += pow(2, i);

	s += rec(0, 1);

	std::cout << s;
}

File: 0671.py
N=input()
arr=[]
s=-1

def f(p, r):
    if not r:
        return 2**(len(N)-p)
    if p==len(N):
        return 1
    t=0
    if arr[p]>4:
        t+=f(p+1, 0)
    if arr[p]>7:
        t+=f(p+1, 0)
    if arr[p]==4 or arr[p]==7:
        t+=f(p+1, 1)
    return t

for i in range(len(N)):
    arr.append(int(N[i]))
    s+=2**i

print(s+f(0, 1))
